patch/patcher.go

```diff
@@ -531,6 +532,22 @@ func (p *Patcher) patchIdent(id *ast.Ident, obj types.Object) {
 			if v.Tag == nil {
 				v.Tag = &ast.BasicLit{}
 			}
+
+			if jsonTag, exist := reflect.StructTag(tags).Lookup("json"); exist {
```

Comment:
This really shouldn’t special-case `json`. If the behavior is to override existing tags, then that behavior should be consistent for all tags, not just `json`.